src/commands/importFromRegistry.ts

```diff
@@ -0,0 +1,246 @@
+import * as vscode from "vscode";
+import { Keychain } from "../keychain";
+
+// To avoid overhead querying the registry, cache responses (each time the command is run)
+const regQueryCache = new Map<string, any>();
+
+export async function importFromRegistry(scope?: vscode.ConfigurationScope) {
+  const config = vscode.workspace.getConfiguration("intersystems", scope);
+  const serverDefinitions: any = config.get("servers");
+
+  const newServerNames = new Array<string>();
+  const serversMissingUsernames = new Array<string>();
+
+  regQueryCache.clear();
+
+  return vscode.window.withProgress({
+    "location": vscode.ProgressLocation.Notification,
+    "cancellable": true
+  }, async (progress, cancellationToken) => {
+    progress.report({message: "Loading server definitions from Windows registry..."});
+    cancellationToken.onCancellationRequested(() => {
+      vscode.window.showInformationMessage("Cancelled server import.");
+    });
+
+    // This forces the progress bar to actually show before the possibly long-running load of registry data
+    await new Promise(resolve => setTimeout(resolve,0));
+
+    await loadRegistryData(config, serverDefinitions, serversMissingUsernames, newServerNames);
+
+    if (cancellationToken.isCancellationRequested) {
+      return false;
+    }
+    return true;
+  }).then(async (keepGoing) => {
+    if (!keepGoing) {
+      return;
+    }
+    if (!await promptForUsernames(serverDefinitions, serversMissingUsernames)) {
+      vscode.window.showInformationMessage("Cancelled server import.");
+      return;
+    }
+    await promptForPasswords(serverDefinitions, newServerNames);
+    await config.update(`servers`, serverDefinitions, vscode.ConfigurationTarget.Global)
+      .then(() => {
+        return vscode.window.showInformationMessage("Server import completed.");
+      }, (reason) => {
+        let message = "Something went wrong importing servers.";
+        if (reason instanceof Error) {
+          message = reason.message;
+        }
+        vscode.window.showErrorMessage(message);
+      });
+  });
+}
+
+async function loadRegistryData(config, serverDefinitions, serversMissingUsernames, newServerNames): Promise<void> {
+  const cmd = require("node-cmd");
+  const hkeyLocalMachine = "HKEY_LOCAL_MACHINE";
+  preloadRegistryCache(cmd, "HKEY_CURRENT_USER\\Software\\InterSystems\\Cache\\Servers");
+  for (const folder of ['','\\WOW6432Node']) {
+    const subFolder = "\\Intersystems\\Cache\\Servers";
+    const path = hkeyLocalMachine + "\\SOFTWARE" + folder + subFolder;
+    preloadRegistryCache(cmd, path);
+    const regData = cmd.runSync("reg query " + path);
```

Comment:
On my system the first pass through this for-loop (path='HKEY_LOCAL_MACHINE\SOFTWARE\Intersystems\Cache\Servers') results in this regData:
```
{data: null, err: 'ERROR: The system was unable to find the specified registry key or value.
', stderr: 'ERROR: The system was unable to find the specified registry key or value.
'}
```

Consequently the next line fails (Cannot read property 'split' of null)

Reply:
@gjsjohnmurray I've tried to avoid messing with my registry in testing for this - should have the fix in at this point, but if there are further issues I'll bite the bullet.